app/services/label/base_label.rb

```diff
@@ -0,0 +1,35 @@
+require 'easyzpl'
+require 'open3'
+
+module Label
+  class BaseLabel
+    attr_accessor :label, :file, :dots, :width, :field_orientation, :height, :print_count
+
+    def initialize(dots, width, field_orientation, height)
+      self.print_count = nil
+      self.label       = Easyzpl::Label.new(dots: dots,
+                                            width: width,
+                                            field_orientation: field_orientation,
+                                            height: height)
+    end
+
+    def tmp_label_file
+      @file = Tempfile.new("cupsjob")
```

Comment:
```suggestion
      @file = Tempfile.new("cupsjob") do |file|
        file.write(label.to_s)
      end
```

Using a block automatically closes the file.